### ISSUE TITLE
[Test failures] FSInternetRadioHandlerOSGiTest again

### ISSUE THREAD

**Issue by sjka** [eclipse/smarthome#3377]
Travis hit this in (completely unrelated #3371):

```
verify OFFLINE Thing-status and NULL Item-state when the HTTP response cannot be parsed correctly(org.eclipse.smarthome.binding.fsinternetradio.test.FSInternetRadioHandlerOSGiTest)  Time elapsed: 10.605 sec  <<< FAILURE!
java.lang.AssertionError: The item's state was not updated correctly when the HTTP response cannot be parsed
Expected: is <NULL>
     but: was <0>
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:865)
	at org.junit.Assert$assertThat$0.callStatic(Unknown Source)
	at org.eclipse.smarthome.binding.fsinternetradio.test.FSInternetRadioHandlerOSGiTest$_verify_OFFLINE_Thing-status_and_NULL_Item-state_when_the_HTTP_response_cannot_be_parsed_correctly_closure4.doCall(FSInternetRadioHandlerOSGiTest.groovy:200)
	at org.eclipse.smarthome.binding.fsinternetradio.test.FSInternetRadioHandlerOSGiTest$_verify_OFFLINE_Thing-status_and_NULL_Item-state_when_the_HTTP_response_cannot_be_parsed_correctly_closure4.doCall(FSInternetRadioHandlerOSGiTest.groovy)
	at sun.reflect.GeneratedMethodAccessor31.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:90)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:324)
	at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:278)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1016)
	at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:39)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:45)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:108)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:112)
	at org.eclipse.smarthome.test.OSGiTest.waitForAssert(OSGiTest.groovy:220)
	at sun.reflect.GeneratedMethodAccessor8.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrap.invoke(PogoMetaMethodSite.java:187)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:56)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:49)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:61)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:153)
	at org.eclipse.smarthome.test.OSGiTest.waitForAssert(OSGiTest.groovy:191)
	at org.eclipse.smarthome.test.OSGiTest.waitForAssert(OSGiTest.groovy)
	at sun.reflect.GeneratedMethodAccessor7.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:207)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:56)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:49)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:133)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:141)
	at org.eclipse.smarthome.binding.fsinternetradio.test.FSInternetRadioHandlerOSGiTest.verify OFFLINE Thing-status and NULL Item-state when the HTTP response cannot be parsed correctly(FSInternetRadioHandlerOSGiTest.groovy:199)

```

**Comment by sjka** [comment]
@m-memova could you please have a look again? To me it looks like the assertion that the item state is `UndefType.NULL` is only correct by chance: If it never was set, then it is right. However, if due to lucky timing it was set, then it stays at this value. The ThingHandler actually does _not_ nullify item states in case of an invalid response. Therefore this assertion is wrong.

...or I am 😉

**Comment by m-memova** [comment]
Thank you, @sjka. It seems that you are right. We can be sure only that the thing will go offline when an exception is thrown. I will remove the item-state-assertion. Are you OK with that?

**Comment by sjka** [comment]
To my understanding it's just a matter of timing. 
Anyway, it would be great if you could remove all of the null-item-state-assertions. It's not a feature of the handler to set them to null in case of an error, so it is consistent to also not check for it in a test-case which targets the handler. 
Thanks for taking care of it!

**Comment by maggu2810** [comment]
Hopefully solved by #3398
